llama-index-core/llama_index/core/workflow/workflow.py

```diff
@@ -275,6 +282,17 @@ async def _task(
                     )
                 )
 
+            # add dedicated cancel task
+            async def _cancel_workflow_task() -> None:
+                await ctx._cancel_flag.wait()
+                raise WorkflowCancelledByUser
+
+            ctx._tasks.add(
+                asyncio.create_task(
+                    _cancel_workflow_task(), name="cancel_workflow_task"
```

Comment:
Interesting method! Basically, this allows us to reuse all our existing logic for canceling tasks 💪🏻

Reply:
Yeah! And it also I think matches the current working relationship between `Worfklow` and `WorkflowHandler`.